.github/workflows/build_rust_docs.yml

```diff
@@ -0,0 +1,28 @@
+name: build-rust-docs
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-rust-docs:
+    runs-on: ubuntu-latest
+    env:
+      RUSTDOCFLAGS: -D warnings
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup toolchain
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v2
+        with:
+          shared-key: cache-dependencies-rustdocs
+        
+      - name: Build docs
+        run: |
+          SQLX_OFFLINE=true cargo doc --no-deps --workspace
```

Comment:
Looking at the original issue (#621), I think we need to add `RUSTDOCFLAGS="-D warnings"` here, otherwise the doc building succeeds even when there are warnings when building the docs.

Reply:
Nvm, it's set up in the general env vars

```
    env:
      RUSTDOCFLAGS: -D warnings
```